<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1746</width>
    <height>1239</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Pulse generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0">
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Pulse configuration</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,1">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="pushButton_add_col_right">
                <property name="text">
                 <string>add column to the right</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_add_col_left">
                <property name="text">
                 <string>add column to the left</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_move_left">
                <property name="text">
                 <string>move column to the left</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_move_right">
                <property name="text">
                 <string>move column to the right</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_invert">
                <property name="text">
                 <string>invert row</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_swap_rows">
                <property name="text">
                 <string>swap selected rows</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_del_col">
                <property name="text">
                 <string>delete column</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_load_config">
                <property name="text">
                 <string>load config</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_extrract_matrix">
                <property name="text">
                 <string>save config</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTableWidget" name="tableWidget">
              <property name="rowCount">
               <number>10</number>
              </property>
              <property name="columnCount">
               <number>1</number>
              </property>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <row/>
              <column/>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="pushButton_add_var_down">
                <property name="text">
                 <string>add parameter</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_save_var">
                <property name="text">
                 <string>save config</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_load_var">
                <property name="text">
                 <string>load config</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTableWidget" name="tableWidget_var">
              <property name="rowCount">
               <number>1</number>
              </property>
              <property name="columnCount">
               <number>4</number>
              </property>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <row/>
              <column>
               <property name="text">
                <string>name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>value</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>description</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>variable</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="PlotWidget" name="pulse_view" native="true"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Measurement parameters</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5" stretch="1,1,1,3,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,0,0,0,0,0,0,0,0,0,0,0,0">
          <property name="topMargin">
           <number>18</number>
          </property>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Variable parameters : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>begin at</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_min">
            <property name="maximum">
             <number>1000000000</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>end at</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_max">
            <property name="maximum">
             <number>1000000000</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>with a step of</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_step">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000000000</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>or</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_num_points">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000000000</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>points</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_plot_pulse">
            <property name="text">
             <string>plot measurement range</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_pulse_sequence">
            <property name="text">
             <string>preview 10 measurement points</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Trigger per point :</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_trigger_per_point_check">
            <property name="text">
             <string>Enabled</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>channel :</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_trigger_per_point_channel"/>
          </item>
          <item>
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>duration (ns):</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_trigger_per_point_duration">
            <property name="maximum">
             <number>100000000</number>
            </property>
            <property name="value">
             <number>10000</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Number of repetition per measurement point:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_n_repeat">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Trigger per sequence (at the begining of the measurement sequence):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>channel :</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_trigger_per_sequence_channel"/>
          </item>
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>duration (ns):</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_trigger_per_sequence_duration">
            <property name="maximum">
             <number>100000000</number>
            </property>
            <property name="value">
             <number>10000</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Number of averaged curved:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_n_average">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="PlotWidget" name="pulse_sequence_view" native="true"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QPushButton" name="pushButton_compute_sequence">
            <property name="text">
             <string>compute final sequence</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBar">
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Number of tupple sent to the PulseStreamer : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_num_tupple">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Total measurement time : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_total_time">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Data acquisition</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6" stretch="1,1,3,3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QPushButton" name="pushButton_start_acquisition">
              <property name="text">
               <string>start acquisition</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_stop_acquisition">
              <property name="text">
               <string>stop acquisition</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="label_live_data">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="PlotWidget" name="plotwidget_live_data" native="true"/>
          </item>
          <item>
           <widget class="PlotWidget" name="plotwidget_DAQ_data" native="true"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1746</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_pulse_sequence"/>
   </widget>
   <widget class="QMenu" name="menuConfig">
    <property name="title">
     <string>Config</string>
    </property>
    <widget class="QMenu" name="menuHardware_configuration">
     <property name="title">
      <string>Hardware configuration</string>
     </property>
     <addaction name="actionpulseStreamer"/>
     <addaction name="actionMFLI"/>
     <addaction name="actionKeysight_33512B"/>
    </widget>
    <addaction name="menuHardware_configuration"/>
    <addaction name="actionSave_pulse_config"/>
    <addaction name="actionLoad_pulse_configuration"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuConfig"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew_pulse_sequence">
   <property name="text">
    <string>New pulse sequence</string>
   </property>
  </action>
  <action name="actionpulseStreamer">
   <property name="text">
    <string>pulseStreamer</string>
   </property>
  </action>
  <action name="actionMFLI">
   <property name="text">
    <string>MFLI</string>
   </property>
  </action>
  <action name="actionKeysight_33512B">
   <property name="text">
    <string>Keysight 33512B</string>
   </property>
  </action>
  <action name="actionSave_pulse_config">
   <property name="text">
    <string>Save pulse configuration</string>
   </property>
  </action>
  <action name="actionLoad_pulse_configuration">
   <property name="text">
    <string>Load pulse configuration</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
